File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/opmodes/AzusaDebug.java
package org.firstinspires.ftc.teamcode.opmodes;

import com.qualcomm.robotcore.eventloop.opmode.TeleOp;

import org.firstinspires.ftc.teamcode.control.path.Path;
import org.firstinspires.ftc.teamcode.control.path.PathPoints;
import org.firstinspires.ftc.teamcode.control.path.builders.PathBuilder;
import org.firstinspires.ftc.teamcode.control.system.Robot;
import org.firstinspires.ftc.teamcode.util.Debuggable;
import org.firstinspires.ftc.teamcode.util.Pose;

@Debuggable
@TeleOp
public class AzusaDebug extends Robot {

    @Override
    public Pose startPose() {
        return new Pose(-64,-64,Math.PI / 2);
    }

    @Override
    public Path path() {
        PathBuilder exp = new PathBuilder("lCurve exp")
                .addPoint(new PathPoints.BasePathPoint("start", -64,-64,0))
                .addPoint(new PathPoints.BasePathPoint("fw", -64, -34, 14))
                .addPoint(new PathPoints.BasePathPoint("joint 1", -46, -10, 14))
                .addPoint(new PathPoints.BasePathPoint("joint 2", -16, 8, 14))
                .addPoint(new PathPoints.BasePathPoint("track 1", 16, 12, 14))
                .addPoint(new PathPoints.StopPathPoint("track 2", 56, 12, 0, 14));
        return exp.build();
    }
}

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/opmodes/AzusaDebug.kt
package org.firstinspires.ftc.teamcode.opmodes

import com.qualcomm.robotcore.eventloop.opmode.Disabled
import com.qualcomm.robotcore.eventloop.opmode.TeleOp
import org.firstinspires.ftc.teamcode.control.path.Path
import org.firstinspires.ftc.teamcode.control.path.StopWaypoint
import org.firstinspires.ftc.teamcode.control.path.Waypoint
import org.firstinspires.ftc.teamcode.control.path.builders.PathBuilder
import org.firstinspires.ftc.teamcode.control.system.AzusaDeprecated
import org.firstinspires.ftc.teamcode.util.*

@Debuggable
@TeleOp
@Deprecated("deprecated with azusa")
@Disabled
class AzusaDebug : AzusaDeprecated() {
    override fun startPose() = Pose(Point(-64.0, -64.0), Angle(Math.PI / 2, AngleUnit.RAD))

    override fun path(): Path {
        val exp = PathBuilder()
            .addPoint(Waypoint("start", -64.0, -64.0, 0.0))
            .addPoint(Waypoint("fw", -64.0, -34.0, 14.0))
            .addPoint(Waypoint("joint 1", -46.0, -10.0, 14.0))
            .addPoint(Waypoint("joint 2", -16.0, 8.0, 14.0))
            .addPoint(Waypoint("track 1", 16.0, 12.0, 14.0))
            .addPoint(StopWaypoint("track 2", 56.0, 12.0, 14.0, Angle(0.0, AngleUnit.RAW)))
        return exp.build()
    }
}
